src/main/java/org/folio/service/processing/ranking/ScoreService.java

```diff
@@ -0,0 +1,140 @@
+package org.folio.service.processing.ranking;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NavigableSet;
+import java.util.Optional;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.dao.DataImportQueueItemDao;
+import org.folio.rest.jaxrs.model.DataImportQueueItem;
+import org.folio.rest.jaxrs.model.DataImportQueueItemCollection;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ScoreService {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  @Autowired
+  private QueueItemHolisticRanker ranker;
+
+  @Autowired
+  private DataImportQueueItemDao queueItemDao;
+
+  /**
+   * Get the next best queue item that is currently waiting to execute
+   */
+  public Future<Optional<DataImportQueueItem>> getBestQueueItem() {
+    return this.getRankedQueueItems()
+      .map((NavigableSet<DataImportQueueItem> set) -> {
+        if (set.isEmpty()) {
+          return Optional.empty();
+        }
+        return Optional.of(set.first());
+      });
+  }
+
+  /**
+   * Get a list of all WAITING queue items, sorted by score, with the highest
+   * score coming first
+   */
+  public Future<NavigableSet<DataImportQueueItem>> getRankedQueueItems() {
```

Comment:
I was trying to think through how this function would work with multiple threads calling it.  Do we have multiple workers calling it?  Does the process of sending a file off to data import require parallel execution?  This would preclude the need for tracking in progress items to load balance tenants(unless we loop back around to tracking items in actual progress with further status). 

I guess this was a long winded way to say we might not need the usage scoring.

Reply:
I'm not sure of the best way to make this concurrent; perhaps I should add some functionality to "claim" a job (mark it as in-progress) and make this entire process atomic, such that only one worker could do it at a time.

Reply:
I'll work on that tomorrow; I also planned on adding some more test cases